source/extensions/filters/network/thrift_proxy/metadata.h

```diff
@@ -57,6 +58,16 @@ class MessageMetadata {
   const Http::HeaderMap& headers() const { return headers_; }
   Http::HeaderMap& headers() { return headers_; }
 
+  /**
+   * @return SpanList an immutable list of Spans
+   */
+  const SpanList& spans() const { return spans_; }
+
+  /**
+   * @return SpanList* a pointer to a mutable list of Spans
```

Comment:
maybe a note here about locking/threading concerns with the returned `SpanList*` ptr

Reply:
There aren't any because these are only ever manipulated or read in the context of an event (network, timer, etc) and those events are serialized by libevent.